client/jetpack-connect/test/plans.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @format
+ * @jest-environment jsdom
+ */
+/**
+ * External dependencies
+ */
+import { mount, render } from 'enzyme';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import PlansWrapper, { getSitePlans, SELECTED_SITE, SITE_PLAN_PRO } from './lib/plans';
+import { PLAN_JETPACK_BUSINESS } from 'lib/plans/constants';
+
+jest.mock( 'components/data/query-plans', () => require( 'components/empty-component' ) );
+jest.mock( 'components/data/query-site-plans', () => require( 'components/empty-component' ) );
+jest.mock( 'jetpack-connect/happychat-button', () => require( 'components/empty-component' ) );
+jest.mock( 'my-sites/plan-features', () => require( 'components/empty-component' ) );
+
+describe( 'Plans', () => {
+	test( 'should render with no plan (free)', () => {
+		const wrapper = render( <PlansWrapper /> );
+
+		expect( wrapper ).toMatchSnapshot();
+	} );
+
+	test( 'should render with a paid plan', () => {
```

Comment:
I'm not sure about this test, because it seems the [intention is to redirect](https://github.com/Automattic/wp-calypso/blob/33c6425d634bca537e2e0bc0b57bef9bb1fe3137/client/jetpack-connect/plans.jsx#L88) for users with paid plans.

Reply:
That's in `componentDidUpdate` which isn't called on initial render. It's hard to determine if that's intentional or an oversight.

If one of our main goals is to describe behavior to detect regressions as we work, I think this test is important.

https://github.com/Automattic/wp-calypso/blob/33c6425d634bca537e2e0bc0b57bef9bb1fe3137/client/jetpack-connect/plans.jsx#L82-L88

Reply:
Pretty sure it is an oversight. Agree about detecting regressions though, so right to keep it for now.